pkg/lockfile/match-build-gradle.go

```diff
@@ -77,4 +82,25 @@ func (m BuildGradleMatcher) Match(sourcefile DepFile, packages []PackageDetails)
 	return nil
 }
 
+/*
+This is based on https://docs.gradle.org/current/userguide/dependency_configurations.html#sub:what-are-dependency-configurations
+We extract a runtimeClasspath scope when we find a runtime only instruction because it will only appear as "testRuntimeClasspath" in the lockfile
+
+This let us make the difference between a testRuntime dependency and a runtime only dependency
+*/
+func (m BuildGradleMatcher) extractingScope(line string) string {
```

Comment:
nit (non-blocking): I think it makes more sense to call this `extractScope` instead